en/animation/marionette/animation-graph-basics.md

```diff
@@ -0,0 +1,221 @@
+
+- **Threshold**: this item is visible when an animation has been added to the animation blending. Each time an animation is added, a new threshold will be added to the lower panel, and you can set each threshold size manually as needed; you can also check **Automatic Threshold Adjustment** to automatically divide the thresholds evenly, either one of the two. All the thresholds are arranged from top to bottom in the order of smallest to largest.
+
+- Animation clips can be attached directly on the animation in the grid layout area, or you can specify the animation clips in the **Inspector** tab on the left after selecting the animation.
+
+  > **Note**: if no animation clip is specified, it will cause problems.
+
+  ![choose clip](animation-graph-basics/choose-clip.png)
+
+  If you need to remove the animation, right-click the animation directly in the grid layout area to remove it, or select the animation and click the gear icon button in the **Inspector** tab on the left to remove it.
+
+For example, to achieve an even blend of standby, walk, and run animations based on the character's rate, set up as the following image:
+
+![blend-speed](animation-graph-basics/blend1-speed.png)
+
+### 2D Animation Blending
+
+**2D animation blending** is the blend of multiple animations based on **two input parameters** by **barycentric interpolation blending algorithm**, each of which can be configured with a threshold value.
+
+The basic operation of 2D animation blending is basically the same as 1D animation blending, the differences are:
+- 2D animation blending has two **parameters** on the **Inspector** tab.
+- The thresholds are represented as `[x, y]` and are ordered top-down according to the order in which the animations are added. and there is no **Automatic Threshold Adjustment** option.
+- The value of the first input parameter corresponds to the interval of the lower threshold `X`, and the value of the second input parameter corresponds to the interval of the lower threshold `Y`.
+
+2D animation blending is often used to blend moving animations in different directions by interpolation, e.g.
+
+![animation-blend 2D](animation-graph-basics/blend2.png)
+
+The range of the first input parameter in the above figure is `[-1, 0]` and the range of the second input parameter is `[-1, 1]`.
+
+## Sub-State Machine
+
+Because objects will typically have complex combinations of actions in multiple stages, it is also possible to (recursively) nest **sub-state machines** within the state machine in order to degrade complex logic. We generally refer to the topmost state machine of the animation graph as the **top-level state machine**, similar to the root node in the scene.
```

Comment:
```suggestion
Since objects will typically have complex combinations of actions in multiple stages, it is also possible to (recursively) nest **sub-state machines** within the state machine in order to degrade complex logic. This is generally referred to as the topmost state machine of the animation graph as the **top-level state machine**, similar to the root node in the scene.
```